workflows/miassembler.nf

```diff
@@ -77,32 +77,46 @@ workflow MIASSEMBLER {
 
     ch_versions = Channel.empty()
 
-    if (params.samplesheet) {
-        groupReads = { study_accession, reads_accession, fq1, fq2, library_layout, library_strategy ->
+    fetch_tool_metadata = Channel.empty()
+
+    if ( params.samplesheet ) {
+
+        groupReads = { study_accession, reads_accession, fq1, fq2, library_layout, library_strategy, assembler, assembly_memory ->
             if (fq2 == []) {
                 return tuple(["id": reads_accession,
                               "study_accession": study_accession,
                               "library_strategy": library_strategy,
                               "library_layout": library_layout,
-                              "single_end": true],
-                             [fq1])
-            }
-            else {
+                              "single_end": true,
+                              "assembler": assembler ?: params.assembler,
```

Comment:
Should we define default assembler to metaspades in nextflow.config? or make assembler required in samplesheet?

Reply:
Commenting without testing - I guess this should still be defined by the complex rules we wrote to pick the assembler automatically based on data?

Reply:
the rule still stands:

1. the samplesheet assembler will be the first option
2. if there is no selected assembler, and it's single end -> megahit
3. if there is no selected assembler, and it's paired end -> metaspades

Reply:
In samplesheet you can skip assembler and in params you can also skip it because it is optional. So in that code you can end up without any assembler chosen.. That is my point